src/compensations/utils/salary.ts

```diff
@@ -0,0 +1,54 @@
+import {
+  isSalariesType,
+  Salaries,
+} from "studio/components/salariesInput/utils/parseSalaries";
+import { SalariesByLocation } from "../../../studio/lib/payloads/compensations";
+import { Result, ResultError, ResultOk } from "../../../studio/utils/result";
+
+export function calculateSalary(
+  examinationYear: number,
+  degree: string,
+  salaries: Salaries,
+): number | undefined {
+  const degreeValue = degree === "bachelor" ? 1 : 0;
+  const adjustedYear = examinationYear + degreeValue;
+  return salaries[adjustedYear];
+}
+
+export function calculatePension(salary: number): number {
+  return Math.round(salary * 0.07);
+}
+
+function examinationYearsFromSalaries(salaries: Salaries) {
+  return Object.keys(salaries).map(Number);
+}
+
+export function minSalariesExaminationYear(salaries: Salaries): number {
+  return Math.min(...examinationYearsFromSalaries(salaries));
+}
+
+export function maxSalariesExaminationYear(salaries: Salaries): number {
+  return Math.max(...examinationYearsFromSalaries(salaries));
+}
+
+export function salariesFromLocation(
+  year: number,
+  locationId: string,
```

Comment:
Better to use location reference instead of this?

Reply:
We could, but the `locations` given to `<Compensations />` are of type `CompanyLocation`, not `Reference`. So we would need to convert `CompanyLocation` to `Reference`, i.e. adding `_type`, and then not use `_type` (just `_ref`) 🙃

Reply:
Perhaps we need some documentation here to avoid confusion for other devs?